src/Control/Effect/Cut.hs

```diff
@@ -1,4 +1,14 @@
 {-# LANGUAGE DeriveFunctor, ExistentialQuantification, FlexibleContexts, GeneralizedNewtypeDeriving, StandaloneDeriving #-}
+
+{- | Provides an effect to delimit backtracking in a given nondeterministic context. This effect is used in concert with 'Control.Effect.Choose.Choose' or 'Control.Effect.NonDet.NonDet'.
+
+Computations that signal failure with 'cutfail' prevent backtracking within the nearest enclosing 'call'.
```

Comment:
I really think we should document this sort of thing on the methods and avoid repeating ourselves. (I’m trying to tackle more of that with #239, but it’s slow going.)